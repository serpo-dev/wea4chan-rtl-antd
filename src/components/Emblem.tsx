import { Col, Row, Typography } from "antd";
import { FC } from "react";
import ChanPNG from "../assets/logo.png";
import { Link } from "react-router-dom";

const { Title } = Typography;

const emblemStyle = {
    height: "100%",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
};

const imgStyle = {
    height: "100%",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    marginLeft: "20px",
    marginRight: "20px",
};

export function Emblem(): ReturnType<FC> {
    return (
        <Row style={emblemStyle}>
            <Link
                style={imgStyle}
                to="/"
            >
                <img
                    style={{ height: "80%" }}
                    className="unselectable undraggabe"
                    src={ChanPNG}
                    alt="logo_chan_png"
                />
            </Link>
            <Link to="/">
                <Title
                    level={2}
                    style={{
                        color: "rgb(4, 119, 135)",
                        margin: "0",
                    }}
                    className="unselectable undraggabe"
                >
                    wea4chan
                </Title>
            </Link>
        </Row>
    );
}
