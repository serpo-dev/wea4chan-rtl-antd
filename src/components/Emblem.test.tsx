import { screen, render, waitFor, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Emblem } from "./Emblem";
import { BrowserRouter } from "react-router-dom";

describe("should render a valid emblem component", () => {
    it("should match a snapshot", () => {
        const { asFragment } = render(<Emblem />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render the icon image", () => {
        render(<Emblem />, { wrapper: BrowserRouter });
        const icon = screen.getByTestId("redirect_icon");
        expect(icon).toBeInTheDocument();
    });

    it("should redirect to the '/' by clicking on the icon image", async () => {
        render(<Emblem />, { wrapper: BrowserRouter });
        const redirectIcon: HTMLAnchorElement =
            screen.getByTestId("redirect_icon");
        expect(redirectIcon.pathname).toEqual("/");
    });

    it("should render the title", () => {
        render(<Emblem />, { wrapper: BrowserRouter });
        const icon = screen.getByTestId("redirect_title");
        expect(icon).toBeInTheDocument();
    });

    it("should redirect to the '/' by clicking on the title", async () => {
        render(<Emblem />, { wrapper: BrowserRouter });
        const redirectTitle: HTMLAnchorElement =
            screen.getByTestId("redirect_title");
        expect(redirectTitle.pathname).toEqual("/");
    });
});
