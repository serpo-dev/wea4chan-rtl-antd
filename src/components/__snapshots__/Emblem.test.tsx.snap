// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a valid emblem component should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-row css-dev-only-do-not-override-12jzuas"
    style="height: 100%; display: flex; align-items: center; justify-content: center;"
  >
    <a
      data-testid="redirect_icon"
      href="/"
      style="height: 100%; display: flex; align-items: center; justify-content: center; margin-left: 20px; margin-right: 20px;"
    >
      <img
        alt="logo_chan_png"
        class="unselectable undraggabe"
        src="logo.png"
        style="height: 80%;"
      />
    </a>
    <a
      data-testid="redirect_title"
      href="/"
    >
      <h2
        class="ant-typography unselectable undraggabe css-dev-only-do-not-override-12jzuas"
        style="color: rgb(4, 119, 135); margin: 0px;"
      >
        wea4chan
      </h2>
    </a>
  </div>
</DocumentFragment>
`;
